### PR TITLE
28 add readme for all services

### LINKED ISSUE #28: Add README for all services

### PR DESCRIPTION
**Pull request by Keyshal-211**
## Description of changes
Please provide a short description of the change or if the change is sufficiently described in the GitHub issue, provide a link to the issue.

  
## How has this been tested?
Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.
In case of automated tests, the test coverage should be reasonably high, especially on Service classes and other classes with complex logic.
    
## Checklist before requesting a review
- [X] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] My code fulfills all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [X] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

**Review by myluki2000 — CHANGES_REQUESTED** [review]
I have already added a readme for the skilllevel service as you can see. Maybe you want to merge my readme and yours.

**Comment by Keyshal-211** [comment]
> I have already added a readme for the skilllevel service as you can see. Maybe you want to merge my readme and yours.

I merged mines and yours, by mistake i removed that. sorry

File: README.md [hunk 1]
```diff
@@ -1,6 +1,18 @@
 # SkillLevel Service
 
-This service is responsible for calculating and storing the skill levels students gain by completing assessments.
+The Skill Level Service, a pivotal component of the "intelligent tutor" system, plays a central role in comprehensively analyzing and enhancing students' learning experiences. This service leverages the renowned "Bloom's Taxonomy" framework to assess and categorize students' cognitive abilities across various dimensions, including their capacity to remember, understand, apply, and analyze course material. These distinct cognitive skills, referred to as "skills" within the system, are evaluated independently for each chapter of a course.
+
+The calculated skill level, expressed as a score ranging from 0 to 100, represents a dynamic measure of a student's proficiency and competence. It is not a static assessment but rather a continuously evolving reflection of a student's learning journey within the platform.[SkillLevel Service Scoring System and Bloom's Taxonomy](https://gits-enpro.readthedocs.io/en/latest/dev-manuals/gamification/Scoring%20System.html)
```
Comment (PaulBredl) [inline README.md]: Its from 0 to 10.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,18 @@
...
+
+The Skill Level Service adeptly harnesses data from various sources, including solved quizzes, content revisitation, engagement with flashcards, and other platform activities, to precisely compute the skill level of each student. By assimilating this multifaceted data, it generates a comprehensive profile that reflects a student's cognitive growth and aptitude within the course.
+
+Extending its reach beyond the boundaries of the platform, this service provides an Application Programming Interface (API) that external services can utilize to contribute to a student's skill level within IT-REX. For instance, when a student successfully completes an exercise on an external platform, this accomplishment seamlessly integrates into the skill level system of IT-REX, providing a holistic view of the student's abilities.
```
Comment (PaulBredl) [inline README.md]: What are you talking about???
We neither provide any way for outside tools to interact nor it was ever a requirement?

File: README.md [hunk 3]
```diff
@@ -1,6 +1,18 @@
...
+
+The calculated skill level, expressed as a score ranging from 0 to 100, represents a dynamic measure of a student's proficiency and competence. It is not a static assessment but rather a continuously evolving reflection of a student's learning journey within the platform.[SkillLevel Service Scoring System and Bloom's Taxonomy](https://gits-enpro.readthedocs.io/en/latest/dev-manuals/gamification/Scoring%20System.html)
+
+The Skill Level Service adeptly harnesses data from various sources, including solved quizzes, content revisitation, engagement with flashcards, and other platform activities, to precisely compute the skill level of each student. By assimilating this multifaceted data, it generates a comprehensive profile that reflects a student's cognitive growth and aptitude within the course.
```
Comment (PaulBredl) [inline README.md]: Thats not very accurate

File: README.md [hunk 4]
```diff
@@ -1,6 +1,18 @@
...
+
+Extending its reach beyond the boundaries of the platform, this service provides an Application Programming Interface (API) that external services can utilize to contribute to a student's skill level within IT-REX. For instance, when a student successfully completes an exercise on an external platform, this accomplishment seamlessly integrates into the skill level system of IT-REX, providing a holistic view of the student's abilities.
+
+Moreover, the system offers a human touch by allowing lecturers and tutors to manually input skill level data, particularly relevant for physical exercises and assignments that may not be automatically tracked. This input ensures that all facets of a student's learning journey are accurately captured and analyzed.
```
Comment (PaulBredl) [inline README.md]: What??? No, it doesn't?

File: README.md (hunk 1 above)
Comment (Keyshal-211) [reply README.md]: changed @PaulBredl

File: README.md [hunk 5]
```diff
@@ -1,6 +1,18 @@
...
+
+The Skill Level Service functions as a sophisticated analyzer, meticulously dissecting and understanding each student's individual learning path. Based on a student's skill level score and progress within the course, the service provides tailored recommendations for the next steps in their educational journey, suggesting both content to continue with and areas to focus on for improvement.
```
Comment (PaulBredl) [inline README.md]: This sentence is so generic and does not add any new information

File: README.md [hunk 6]
```diff
@@ -1,6 +1,18 @@
...
+
+In summary, the Skill Level Service combines data-driven analysis with human input, aligning with Bloom's Taxonomy to provide a comprehensive understanding of student proficiency. It empowers both students and educators with valuable insights, fostering a personalized and effective learning experience within IT-REX.
```
Comment (PaulBredl) [inline README.md]: This sentence is so generic and does not add any new information
